### PR TITLE
Feature/3928 add integration tests sending of email

### PR DESCRIPTION
**Pull request by RuthBerry96**
Story: [AB#3928](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/3928)
Task: [AB#9449](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/9449)

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj [hunk 1]
```diff
@@ -1,50 +1,52 @@
...
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.1.7" />
+    <PackageReference Include="NHSD.BuyingCatalogue.EmailClient" Version="2.1.0" />
+    <PackageReference Include="NHSD.BuyingCatalogue.EmailClient.IntegrationTesting" Version="2.1.0" />
+    <PackageReference Include="nunit" Version="3.12.0" />
+    <PackageReference Include="NUnit3TestAdapter" Version="3.17.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.6.1" />
+    <PackageReference Include="SpecFlow" Version="3.3.57" />
+    <PackageReference Include="SpecFlow.NUnit" Version="3.3.57" />
+    <PackageReference Include="SpecFlow.Tools.MsBuild.Generation" Version="3.3.57" />
+    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.7.1" />
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj:46]: Any reason for the introduction of the JWT package?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/EmailSteps.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                    new EmailAddress(Empty, expectedContents.To)
+                }
+            };
+            
+            var actual = (await _emailServerDriver.FindAllEmailsAsync()).First();
+            expected.Should().BeEquivalentTo(actual, conf => conf.Excluding(x => x.Html));
+
+        }
+
+        private sealed class EmailContents
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/EmailSteps.cs:64]: Looking at the other step classes, this "data" class usually ends in 'Table'. Seems to be bit of a convention but I don't think it has been discussed, so welcome to put this comment to one side.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Status.feature [hunk 3]
```diff
@@ -117,3 +117,14 @@ Scenario: 10. Update an order status sets the complete date
...
     And the order completed date is set
+
+@3928
+Scenario: 11. When an order is complete, an email is sent
+    Given the user creates a request to update the order status for the order with ID 'C000014-01'
+    And the user enters the 'order-status-complete' update order status request payload
+    When the user sends the update order status request
+    Then a response with status code 204 is returned
+    And only one email is sent
+    And the email sent contains the following information
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Status.feature:128]: Unfortunately might need to split this assertion out. This email in the future will contain more than one attachment & I am not sure how to do this with the step as it is. I wonder if we can simplify the assertion of the attachment document to check each line. Any ideas? What do you think?

Could be raised as tech debt to get this story in.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/EmailSteps.cs (hunk 2 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/EmailSteps.cs:64]: Happy to change it

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj (hunk 1 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.csproj]: It was actually there before (but due to a line ending change just looked more prominant. I've removed it and re ran tests (and it is no longer needed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Status.feature (hunk 3 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Status.feature:128]: I think raise it as tech debt so we can merge the ordering in. I can raise it if you want. As tbh I did a similar version as when there are more headings ect, you will want to change how it is written

**Review by sam-roberts-bjss — APPROVED** [review]
👍

**Pull request merged**